### ISSUE TITLE
Autocompletion (with readline) doesn't handle spaces in :limited_to values

### ISSUE THREAD

**Issue by maxlinc** [erikhuda/thor#458]
I noticed this while playing around with ideas for #457.

If you have something like this:

``` ruby
api_names = [
  'Cloud Identity - List Users',
  'Cloud Identity - Reset user API key',
  'Cloud Identity - List Credentials',
  'Auto Scale - List scaling groups',
  'Cloud Big Data - View profile information'
]
ask 'Which API would you like to test?', :limited_to => api_names
```

Autocomplete will work for `Auto\t`, since the only match is `Auto Scale - List scaling groups`.
However, if you put in `Clo\t` it will complete to `Cloud`, but then the autocomplete has reset double tabbing shows all available options, and adding `Id\t` won't complete `Identity`. Similarly, if you enter `Cloud Id\t` it will not autocomplete to `Cloud Identity`.

Basically, the problem seems to be that Thor doesn't override (or provide options to manually override) relevant readline settings (e.g. `basic_quote_characters`, `basic_word_break_characters`, `completer_quote_character`, or `completer_word_break_characters`) for autocompletion lists that include multi-word choices.
